File: character_py/character.py
class Character:

    _MIN_LEVEL = 1
    _MAX_LEVEL = 100

    _MIN_NAME_LENGTH = 1
    _MAX_NAME_LENGTH = 10

    def __init__(self) -> None:
        self.name = ""
        self.level = self._MIN_LEVEL

    def set_level(self, new_level) -> None:

        if new_level < self._MIN_LEVEL:
            self.level = self._MIN_LEVEL

        elif new_level > self._MAX_LEVEL:
            self.level = self._MAX_LEVEL

        else:
            self.level = new_level

    def level_up(self) -> None:

        if self.level < self._MAX_LEVEL:
            self.level += 1

    def set_name(self, new_name) -> bool:

        if len(new_name) < self._MIN_NAME_LENGTH or len(new_name) > self._MAX_NAME_LENGTH:
            return False 

        self.name = new_name
        return True

File: character_cpp/src/character.cpp
#include <character/character.hpp>

Character::Character() {
    level_ = MIN_LEVEL;
    name_ = "";
}

// Const members must be initialized, since they are static
// this is done in the .cpp file rather than as part of
// instantiation.
const int Character::MIN_LEVEL = 1;
const int Character::MAX_LEVEL = 100;
const int Character::MIN_NAME_LENGTH = 1;
const int Character::MAX_NAME_LENGTH = 10;

void Character::set_level(int new_level) {
    if (new_level < MIN_LEVEL) {
        level_ = MIN_LEVEL;
    } else if (new_level > MAX_LEVEL) {
        level_ = MAX_LEVEL;
    } else {
        level_ = new_level;
    }
}

int Character::get_level() { return level_; }

void Character::level_up() {
    if (level_ < MAX_LEVEL) {
        level_++;
    }
}

bool Character::set_name(std::string new_name) {
    if (new_name.size() > MAX_NAME_LENGTH ||
        new_name.size() < MIN_NAME_LENGTH) {
        return false;
    }

    name_ = new_name;
    return true;
}

std::string Character::get_name() { return name_; }